### DISCUSSION TITLE
npm test Jasmine

### DISCUSSION THREAD

**Discussion by HenryMigo** [community/community#27025] [Actions]
<p>Hi all,</p>
<p>I generated a Angluar App from the CLI and have started to implement the GitHub Actions CI pipeline to it however everytime I get to <code>npm test</code> it tries to open a browser and fails to do so and then hangs on that action until I cancel it.</p>
<p>Is the solution for this to change testing frameworks? or is there a way to get this working?</p>
<p>Here is my yaml file:</p>
<pre><code>name: Angular App CI

on:
  push:
    branches: [master]
  pull_request:
    branches: [master]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [12.x]

    steps:
    - uses: actions/checkout@v2
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    - run: npm ci
    - run: npm test
      env:
        CI: true
    - run: npm run build --if-present
</code></pre>
<p>Henry</p>

**Comment by Siedlerchr** [comment, marked as answer]
<p>Hi,</p>
<p>I am using chrome-headless to run karma jasmine tests for an angular app:</p>
<p>In my package json I declared:</p>
<pre><code>"test-headless": "ng test --watch=false --browsers=ChromeHeadless",
</code></pre>
<p>and in my yaml file:</p>
<pre><code>- name: run on linux
      if: startsWith(matrix.os,'ubuntu')
      run: |
        npm run test-headless
</code></pre>
<p>For the e2e test you might need to modify the protactor conf as well:</p>
<p><a href="https://medium.com/angular-in-depth/angular-testing-with-headless-chrome-d1343b349699" rel="nofollow noopener">https://medium.com/angular-in-depth/angular-testing-with-headless-chrome-d1343b349699</a></p>
<p>Regards</p>

**Comment by HenryMigo** [comment]
<p>Spot on this solved the issue!</p>

**Comment by dbs-17** [comment]
<blockquote>
<p><a class="mention" href="/siedlerchr">@siedlerchr</a> wrote:</p>
</blockquote>
<blockquote>
<p>Hi,</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>I am using chrome-headless to run karma jasmine tests for an angular app:</p>
</blockquote>
<blockquote>
<p>In my package json I declared:</p>
</blockquote>
<blockquote>
<p>“test-headless”: “ng test --watch=false --browsers=ChromeHeadless”,</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>and in my yaml file:</p>
</blockquote>
<blockquote>
<ul>
<li>name: run on linux<br>
if: startsWith(matrix.os,‘ubuntu’)<br>
run: |<br>
npm run test-headless</li>
</ul>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>For the e2e test you might need to modify the protactor conf as well:</p>
</blockquote>
<blockquote>
<p><a href="https://medium.com/angular-in-depth/angular-testing-with-headless-chrome-d1343b349699" rel="nofollow noopener">https://medium.com/angular-in-depth/angular-testing-with-headless-chrome-d1343b349699</a></p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Regards</p>
</blockquote>

**Comment by mwalter** [comment]
<p>Why is this working for you?</p>
<p>I have a very similar setup but I get the error that Chrome cannot be started due to missing libraries on the ubuntu-latest image. Please have a look at <a href="https://github.community/t/npm-test-says-cannot-start-chromeheadless/18310">https://github.community/t5/GitHub-Actions/npm-test-says-Cannot-start-ChromeHeadless/td-p/56613/jump-to/first-unread-message</a>.</p>
<p>Any help is appreciated!</p>
